File: src/cli/process.go
package cli

import (
	"fmt"
)

// Process wraps process management commands
type Process struct {
	signal         int
	fullMatch      bool
	suppressErrors bool
}

// NewProcess creates a new Process command builder
func NewProcess() *Process {
	return &Process{
		signal:         9,
		suppressErrors: true,
	}
}

// Signal sets signal number
func (p *Process) Signal(value int) *Process {
	p.signal = value
	return p
}

// FullMatch uses full match pattern
func (p *Process) FullMatch(value bool) *Process {
	p.fullMatch = value
	return p
}

// SuppressErrors suppresses errors
func (p *Process) SuppressErrors(value bool) *Process {
	p.suppressErrors = value
	return p
}

// Pkill generates pkill command
func (p *Process) Pkill(pattern string) string {
	flags := fmt.Sprintf("-%d", p.signal)
	if p.fullMatch {
		flags += " -f"
	}
	cmd := fmt.Sprintf("pkill %s %s", flags, quote(pattern))
	if p.suppressErrors {
		cmd += " || true"
	}
	return cmd
}

// LsofFile generates lsof command to find process using a file
func (p *Process) LsofFile(filePath string) string {
	cmd := fmt.Sprintf("lsof -t %s", quote(filePath))
	cmd += " | head -1"
	return cmd
}

// FuserFile generates fuser command to find process using a file
func (p *Process) FuserFile(filePath string) string {
	cmd := fmt.Sprintf("fuser %s", quote(filePath))
	cmd += " | grep -oE '[0-9]+' | head -1"
	return cmd
}

// CheckPID generates command to check if process with PID exists
func (p *Process) CheckPID(pid int) string {
	cmd := fmt.Sprintf("kill -0 %d", pid)
	cmd += " && echo exists || echo not_found"
	return cmd
}

// GetProcessName generates command to get process name by PID
func (p *Process) GetProcessName(pid int) string {
	cmd := fmt.Sprintf("ps -p %d -o comm=", pid)
	cmd += " || echo unknown"
	return cmd
}

// Kill generates kill command for a specific PID
func (p *Process) Kill(pid int) string {
	cmd := fmt.Sprintf("kill -%d %d", p.signal, pid)
	if p.suppressErrors {
		cmd += " || true"
	}
	return cmd
}

File: src/CLI/Process.cs
using System;

namespace Enva.CLI;

public class Process
{
    private int signal = 9;
    private bool fullMatch;
    private bool suppressErrors = true;

    public static Process NewProcess()
    {
        return new Process();
    }

    public Process Signal(int value)
    {
        signal = value;
        return this;
    }

    public Process FullMatch(bool value)
    {
        fullMatch = value;
        return this;
    }

    public Process SuppressErrors(bool value)
    {
        suppressErrors = value;
        return this;
    }

    public string Pkill(string pattern)
    {
        string flags = $"-{signal}";
        if (fullMatch)
        {
            flags += " -f";
        }
        string cmd = $"pkill {flags} {Quote(pattern)}";
        if (suppressErrors)
        {
            cmd += " || true";
        }
        return cmd;
    }

    public string LsofFile(string filePath)
    {
        string cmd = $"lsof -t {Quote(filePath)}";
        cmd += " | head -1";
        return cmd;
    }

    public string FuserFile(string filePath)
    {
        string cmd = $"fuser {Quote(filePath)}";
        cmd += " | grep -oE '[0-9]+' | head -1";
        return cmd;
    }

    public string CheckPID(int pid)
    {
        string cmd = $"kill -0 {pid}";
        cmd += " && echo exists || echo not_found";
        return cmd;
    }

    public string GetProcessName(int pid)
    {
        string cmd = $"ps -p {pid} -o comm=";
        cmd += " || echo unknown";
        return cmd;
    }

    public string Kill(int pid)
    {
        string cmd = $"kill -{signal} {pid}";
        if (suppressErrors)
        {
            cmd += " || true";
        }
        return cmd;
    }

    private string Quote(string s)
    {
        if (s.Contains(" ") || s.Contains("$") || s.Contains("'"))
        {
            return $"'{s.Replace("'", "'\"'\"'")}'";
        }
        return s;
    }
}
